app/scenes/Document/components/MarkAsViewed.js

```diff
@@ -17,7 +17,7 @@ class MarkAsViewed extends React.Component<Props> {
 
     this.viewTimeout = setTimeout(() => {
       if (document.publishedAt) {
-        document.view();
+        document.view().then((view) => document.updateLastViewed(view));
```

Comment:
```suggestion
        const view = await document.view();
        document.updateLastViewed(view);
```

Will need to update function to `async` of course